src/Composer/ComposerWidget.vala

```diff
@@ -631,6 +709,54 @@ public class Mail.ComposerWidget : Gtk.Grid {
         from_combo.active = 0;
     }
 
+    private class Attachment : Gtk.FlowBoxChild {
+        public GLib.FileInfo info { get; construct; }
+        public string uri { get; construct; }
```

Comment:
Why not keep a GLib.File here ? You seem to get them and recreate them afterward
```suggestion
        public GLib.File file { get; construct; }
```